include/fw_info.h

```diff
@@ -78,75 +60,156 @@ struct __packed fw_info {
 				#member " has wrong offset")
 
 /* Static asserts to ensure compatibility */
-OFFSET_CHECK(struct fw_info, magic, 0);
-OFFSET_CHECK(struct fw_info, firmware_size, CONFIG_FW_INFO_MAGIC_LEN);
-OFFSET_CHECK(struct fw_info, firmware_version,
-	(CONFIG_FW_INFO_MAGIC_LEN + 4));
-OFFSET_CHECK(struct fw_info, firmware_address,
-	(CONFIG_FW_INFO_MAGIC_LEN + 8));
+OFFSET_CHECK(struct fw_info_abi, abi_len, 0);
+OFFSET_CHECK(struct fw_info_abi, abi_id, 4);
+OFFSET_CHECK(struct fw_info_abi, abi_flags, 8);
+OFFSET_CHECK(struct fw_info_abi, abi_version, 12);
+/** @endcond
+ */
+
+
+/**
+ * This struct is used to request an ABI. It contains a pointer to a non-
```

Comment:
I would suggest a @details field instead of all the details in one long text

/**
 * Structure used to request an ABI
*
* @details ....
*
* @note If `required` is true the image will not function if it has no access to the ABI.
*/

I would also recommend moving the information about the pointer to a non-initialized pointer to the structure member **abi** instead, to improve readibility. If this is done, the ingress could talk about generic functionality of this struct

Reply:
Added `@details`.